client/src/modules/account_statement/account_statement.service.js

```diff
@@ -0,0 +1,113 @@
+  service.filters = accountStatementFilters;
+  service.openCommentModal = openCommentModal;
+  service.commentAccountStatement = commentAccountStatement;
+
+  /**
+   * @method commentAccountStatement
+   * @param {object} params - { uuids: An array of uuid, string: comment }
+   */
+  function commentAccountStatement(params) {
+    return $http.put(baseUrl.concat('comment'), { params : params })
+      .then(util.unwrapHttpResponse);
+  }
+
+  /**
+   * @method openCommentModal
+   * @param {object} request
+   */
+  function openCommentModal(request) {
+    var params = {
+      templateUrl  : 'modules/account_statement/modals/comment.modal.html',
+      controller   : 'CommentAccountStatementController',
+      controllerAs : '$ctrl',
+      size         : 'md',
+      backdrop     : 'static',
+      resolve : {
+        data :  function dataProvider() { return request; },
+      },
+    };
+    var instance = Modal.open(params);
+    return instance.result;
+  }
+
+  // bind cache methods
+  service.removeFilter = removeFilter;
+  service.cacheFilters = cacheFilters;
+  service.loadCachedFilters = loadCachedFilters;
+
+  // default filtes will always be applied
+  accountStatementFilters.registerDefaultFilters([
+    { key : 'account_id', label : 'TABLE.COLUMNS.ACCOUNT' },
+    { key : 'period', label : 'TABLE.COLUMNS.PERIOD', valueFilter : 'translate' },
+    { key : 'custom_period_start', label : 'PERIODS.START', valueFilter : 'date' },
+    { key : 'custom_period_end', label : 'PERIODS.END', valueFilter : 'date' },
+    { key : 'limit', label : 'FORM.LABELS.LIMIT' }]);
+
+  // custom filters can be optionally applied
+  accountStatementFilters.registerCustomFilters([
+    { key : 'trans_id', label : 'FORM.LABELS.TRANS_ID' },
+    { key : 'reference', label : 'FORM.LABELS.REFERENCE' },
+    { key : 'user_id', label : 'FORM.LABELS.USER' },
+    { key : 'amount', label : 'FORM.LABELS.AMOUNT' },
+    { key : 'project_id', label : 'FORM.LABELS.PROJECT' },
+    { key : 'description', label : 'FORM.LABELS.DESCRIPTION' },
+    { key : 'origin_id', label : 'FORM.LABELS.TRANSACTION_TYPE' }]);
+
+
+  if (filterCache.filters) {
+    // load cached filter definition if it exists
+    accountStatementFilters.loadCache(filterCache.filters);
+  }
+
+  // once the cache has been loaded - ensure that default filters are provided appropriate values
+  assignDefaultFilters();
+
+  function assignDefaultFilters() {
+    // get the keys of filters already assigned - on initial load this will be empty
+    var assignedKeys = Object.keys(accountStatementFilters.formatHTTP());
+
+    // assign default period filter
+    var periodKeys = ['period', 'custom_period_start', 'custom_period_end'];
+    var periodDefined = util.arrayIncludes(assignedKeys, periodKeys);
+
+    if (!periodDefined) {
+      accountStatementFilters.assignFilters(Periods.defaultFilters());
+    }
+
+    // assign default limit filter
+    if (assignedKeys.indexOf('limit') === -1) {
```

Comment:
I think this whole process can be improved, my comment on the previous PR was the the hard-coded default value `100` could be shared across modules with `bhConstant`.